### PR TITLE
[MAINTENANCE] Refactor out `termcolor` dependency

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Use built-in `click` functionality so we can get rid of `termcolor`
    - This seems to be the more robust of the two libraries and we already need `click` for CLI processing.

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/cli/v012/util.py [hunk 1]
```diff
@@ -2,47 +2,35 @@
...
-try:
-    from termcolor import colored
-except ImportError:
-    colored = None
+SUPPORTED_CLI_COLORS: Final[List[str]] = ["blue", "cyan", "green", "yellow", "red"]
 
 
-def cli_message(string) -> None:
+def cli_message(string: str) -> None:
     print(cli_colorize_string(string))
```
Comment (cdkini) [inline great_expectations/cli/v012/util.py:28]: This entire file is duplicating efforts in the primary CLI directory - while this feels bad (and is bad), this code is outdated and will soon be removed from the codebase.

I don't think it is worth the effort to refactor existing code to refactor out this duplication. See a Slack conversation around this for more information: https://greatexpectationslabs.slack.com/archives/C9GDA5SAY/p1668435895377679

File: tests/cli/test_datasource_pandas.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 from great_expectations import DataContext
 from great_expectations.cli import cli
-from tests.cli.utils import assert_no_logging_messages_or_tracebacks
+from tests.cli.utils import assert_no_logging_messages_or_tracebacks, escape_ansi
```
Comment (cdkini) [inline tests/cli/test_datasource_pandas.py:11]: Our testing for CLI output is incredibly granular and is a bit overkill in my opinion. I've added this `escape_ansi` helper to strip expected and actual outputs so we can assert the important textual content. Colors are helpful but I think validating hundreds of ANSI codes doesn't provide any real value.

Let me know if you disagree!

File: great_expectations/cli/pretty_printing.py [hunk 3]
```diff
@@ -2,33 +2,25 @@
 import sys
 from typing import List, Optional
 
-from termcolor import colored
+import click
+from typing_extensions import Final
+
+SUPPORTED_CLI_COLORS: Final[List[str]] = ["blue", "cyan", "green", "yellow", "red"]
```
Comment (Kilo59) [inline great_expectations/cli/pretty_printing.py]: If this is a constant should we make this a `tuple`?

Comment (cdkini) [reply great_expectations/cli/pretty_printing.py]: yeah I like that

Comment (cdkini) [reply great_expectations/cli/pretty_printing.py]: done!

Comment (cdkini) [reply great_expectations/cli/pretty_printing.py]: done!

**Review by Kilo59 — APPROVED** [review]
🚀

File: tests/cli/utils.py [hunk 4]
```diff
@@ -128,3 +129,9 @@ def assert_no_tracebacks(click_result):
     except ValueError as ve:
         # sometimes stderr is not captured separately
         pass
+
+
+def escape_ansi(line: str) -> str:
```
Comment (Kilo59) [inline tests/cli/utils.py:134]: If this get called repeatedly on the same string we could speed up the tests with a `functool.lru_cache` here.

Although it's very possible the difference is negligible. 
```suggestion
@functools.lru_cache(maxsize=64)
def escape_ansi(line: str) -> str:
```

**Pull request merged**